owasp-dep-check/Dockerfile

```diff
@@ -1,76 +1,51 @@
-COPY prebuild/owasp-dep-check-dependencies-$SDP_DCAR_OWASP_DEP_CHK_VERSION.sha256 /root/tmp/.
-
-RUN cd /root/tmp/ && gpg --import BAH-public.key
-RUN cd /root/tmp/ && gpg --verify owasp-dep-check-dependencies-$SDP_DCAR_OWASP_DEP_CHK_VERSION.sig owasp-dep-check-dependencies-$SDP_DCAR_OWASP_DEP_CHK_VERSION.tar.gz
-RUN cd /root/tmp/  && echo "$(cat owasp-dep-check-dependencies-$SDP_DCAR_OWASP_DEP_CHK_VERSION.sha256) owasp-dep-check-dependencies-$SDP_DCAR_OWASP_DEP_CHK_VERSION.tar.gz" | sha256sum --check --status
-
-# Unpack dependencies and install packages
-
-RUN cd /root/tmp && tar -xzf owasp-dep-check-dependencies-$SDP_DCAR_OWASP_DEP_CHK_VERSION.tar.gz && \
-    gem install --force --local /root/tmp/dependencies/bundle-audit/*.gem --no-document --no-rdoc --no-ri && \
-    rm -rf /usr/local/share/gems/gems/rubygems-update-2.7.10/test && \
-    gem cleanup
-
-RUN cd /root/tmp && rpm -ivh --replacepkgs --replacefiles --force  /root/tmp/dependencies/mono-complete/*.rpm
-
-RUN cd /root/tmp/dependencies/owasp && \
-    unzip dependency-check-${OWASP_DEP_CHK_VERSION}-release.zip    && \
-    rm dependency-check-${OWASP_DEP_CHK_VERSION}-release.zip       && \
-    mv dependency-check /usr/share/                                         && \
-    rm -rf /root/tmp/dependencies
-
-# Add user, create required directories  and cleanup
-RUN useradd -ms /bin/bash ${user}                                           && \
-    chown -R ${user}:${user} /usr/share/dependency-check                    && \
-    mkdir /report                                                           && \
-    chown -R ${user}:${user} /report                                        && \
-    yum --nogpgcheck --disableplugin=subscription-manager clean all
+### Update and install required packages
+RUN dnf update -y --nodocs && \
+    dnf install -y java-17-openjdk-devel unzip && \
+    dnf module install -y nodejs:16 && \
+    dnf clean all && \
+    npm install --global yarn && \
+    rm -rf /var/cache/dnf
+
+ENV OWASP_DEP_CHK_VERSION 7.3.0
+
+### Download and install OWASP Dependency Check CLI tool
+RUN curl -sSLo /tmp/dependency-check.zip https://github.com/jeremylong/DependencyCheck/releases/download/v$OWASP_DEP_CHK_VERSION/dependency-check-$OWASP_DEP_CHK_VERSION-release.zip && \
+    cd /usr/share && \
+    unzip /tmp/dependency-check.zip && \
+    rm /tmp/dependency-check.zip
+
+### Add user, create required directories and cleanup
+ARG USER=depcheck
+ARG GROUP=depcheck
+ARG UID=1001
+ARG GID=1001
+
+RUN groupadd -r -g ${GID} ${GROUP} && \
+    useradd -ms /bin/bash -u ${UID} -g ${GID} ${USER} && \
+    chown -R ${UID}:${GID} /usr/share/dependency-check && \
+    mkdir /report && \
+    chown -R ${UID}:${GID} /report
 
 ### Insert Container Entry Point Script
 COPY container-entrypoint.sh /usr/share/dependency-check/bin/container-entrypoint.sh
 
 ### Set script as executable
 RUN chmod +x /usr/share/dependency-check/bin/container-entrypoint.sh
 
-USER ${user}
+USER ${UID}
```

Comment:
Were you able to test the image successfully in the pipeline? I thought I saw mention of local tests, asking because the pipeline runs the image with the docker plugin which defaults to running "docker run -u 0:0" This ends up running the image as root.